lib/resty/rediscluster.lua

```diff
@@ -257,6 +332,22 @@ function _M.refresh_slots(self)
     end
 
     self:fetch_slots()
+    -- Cleanup health dict entries for removed nodes
+    local health_dict = ngx.shared[DEFAULT_HEALTH_DICT_NAME]
+    local current_nodes = {}
+    local servers = slot_cache[self.config.name .. "serv_list"].serv_list
+    for _, node in ipairs(servers) do
+        local key = node.ip .. ":" .. node.port
+        current_nodes[key] = true
+    end
+    -- Cleanup stale nodes
+    local all_keys = health_dict:get_keys()
+    for _, key in ipairs(all_keys) do
+        if not current_nodes[key] then
+            health_dict:delete(key)
+        end
+    end
```

Comment:
Why does the key still only have IP and port? Did you not understand the meaning of this comment? https://github.com/api7/lua-resty-redis-cluster/pull/10#discussion_r2057369774